### PR TITLE
new base

### PR DESCRIPTION
**Pull request by shantanuFyndGit**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,24 @@
-# pixelbin-zap
-Pixelbin.io's Zapier Integration containing actions and triggers. 
+# PixelBinMain
```
Comment (sagarika-fynd) [inline README.md]: rename this to Pixelbin.io Zapier Integration

File: .DS_Store
Comment (sagarika-fynd) [inline .DS_Store:1]: Add this in gitignore

File: authentication.js [hunk 2]
```diff
@@ -0,0 +1,383 @@
+"use strict";
+const zapier = require("zapier-platform-core");
+
+const test = (z, bundle) => {
+	zapier.tools.env.inject();
+	const url = `${process.env.BASE_URL}/service/platform/organization/v1.0/apps/info`;
```
Comment (sagarika-fynd) [inline authentication.js]: y cant we use pixelbin's sdk here ?

File: authentication.js [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+		.then((response) => {
+			return response;
+		})
+		.catch((error) => {
+			console.error("Error while testing:", error);
+			throw error;
+		});
+};
+
+const handleBadResponses = (response, z, bundle) => {
```
Comment (sagarika-fynd) [inline authentication.js:22]: what about other errors ? atleast log this

File: authentication.js [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+		type: "custom",
+
+		fields: [
+			{
+				key: "apiKey",
+				label: "API Key",
+				required: true,
+				type: "password",
+				helpText:
+					"1. Go to [console.pixelbin.io](https://console.pixelbin.io/choose-org?redirectTo=settings/apps) and create a free account (you will need to confirm your email). \n2. You can refer our [documentation](https://www.pixelbin.io/docs/getting-started) to create account. \n3. After that, you can find your API token [here](https://console.pixelbin.io/choose-org?redirectTo=settings/apps).",
```
Comment (sagarika-fynd) [inline authentication.js]: give ap token documentation

File: build/build.zip
Comment (sagarika-fynd) [inline build/build.zip:1]: why is this required ?

File: build/source.zip
Comment (sagarika-fynd) [inline build/source.zip:1]: y is this required ?

File: creates/createReport.js
Comment (sagarika-fynd) [inline creates/createReport.js:1]: rename to createUsageReport

File: creates/createReport.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+const { PixelbinConfig, PixelbinClient } = require("@pixelbin/admin");
+const sample = require("../samples/sample_issue");
```
Comment (sagarika-fynd) [inline creates/createReport.js]: y do u need this sample ?

File: creates/createReport.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+const zapier = require("zapier-platform-core");
+const createIssue = async (z, bundle) => {
```
Comment (sagarika-fynd) [inline creates/createReport.js]: y is this function called createIssue ?

File: creates/createReport.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	return newData;
+};
+
+module.exports = {
+	key: "createReport",
+	noun: "createReport",
+
+	display: {
+		label: "Create Usage Report",
+		description: "Generates your PixelBin.io's storage report",
```
Comment (sagarika-fynd) [inline creates/createReport.js]: Generates your PixelBin.io's storage report => Generates your PixelBin.io's usage report

File: creates/transform.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
+const sample = require("../samples/sample_issue");
```
Comment (sagarika-fynd) [inline creates/transform.js]: y do u need sample ?

File: creates/transform.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+const zapier = require("zapier-platform-core");
+
+const perform = async (z, bundle) => {
+	zapier.tools.env.inject();
+	imagetobeTransformed = "";
```
Comment (sagarika-fynd) [inline creates/transform.js]: use proper camelCase

File: creates/transform.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+	if (bundle.inputData.url.includes(`${process.env.CDN_URL}`)) {
+		imagetobeTransformed = bundle.inputData.url;
+	} else {
+		try {
+			const response = await z.request({
+				url: `${process.env.BASE_URL}/service/platform/assets/v1.0/upload/url`,
```
Comment (sagarika-fynd) [inline creates/transform.js:11]: y cant we use SDK ?

File: creates/transform.js [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+				method: "POST",
+				headers: {
+					accept: "application/json",
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify({
+					url: bundle.inputData.url,
+					path: "/__zapier_Transformation",
```
Comment (sagarika-fynd) [inline creates/transform.js]: path : __zapier/tranformation

File: creates/transform.js [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+				}),
+			});
+			imagetobeTransformed = response.data.url;
+		} catch (error) {
+			throw new Error(`FAILED to upload image: ${error}`);
+		}
+	}
+
+	let replacement = bundle.inputData.transformationString;
+	imagetobeTransformed = imagetobeTransformed.replace("original", replacement);
```
Comment (sagarika-fynd) [inline creates/transform.js]: incase of pixelbin existing image, may not have original keyword in it, you will have to add to existing transformation

File: creates/transform.js [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+			throw error;
+		}
+	}
+
+	return getStatus();
+};
+
+module.exports = {
+	key: "transform",
+	noun: "transform",
```
Comment (sagarika-fynd) [inline creates/transform.js]: noun should be image ?

File: creates/upload.js [hunk 14]
```diff
@@ -0,0 +1,108 @@
+const zapier = require("zapier-platform-core");
+
+const perform = async (z, bundle) => {
+	zapier.tools.env.inject();
+	const bodyData = {
+		url: bundle.inputData.url,
+		path: bundle.inputData.path
+			? `/${bundle.inputData.path}`
+			: "/__zapier_Uploads",
```
Comment (sagarika-fynd) [inline creates/upload.js]: __zapier/Upload

File: creates/upload.js [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+	return responseData;
+};
+module.exports = {
+	key: "uploadFile",
+	noun: "File",
+	display: {
+		label: "Upload File to Pixelbin",
+		description: "Uploads an image to Pixelbin.io.",
```
Comment (sagarika-fynd) [inline creates/upload.js]: explain output will be image url

File: index.js [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+	afters = [],
+} = require("./authentication");
+
+const App = {
+	version: require("./package.json").version,
+	platformVersion: require("zapier-platform-core").version,
+	authentication,
+
+	// beforeRequest & afterResponse are optional hooks into the provided HTTP client
+	beforeRequest: [...befores],
```
Comment (sagarika-fynd) [inline index.js:20]: since u dont have any before and after can you remove this ?

File: package-lock.json
Comment (sagarika-fynd) [inline package-lock.json:1]: is this code need to be commited

File: samples/sample_repo_list.js [hunk 17]
```diff
@@ -0,0 +1,109 @@
+module.exports =   {
```
Comment (sagarika-fynd) [inline samples/sample_repo_list.js]: do you need this ?

File: triggers/createFile.js [hunk 18]
```diff
@@ -0,0 +1,252 @@
+var hookID = "";
+const eventIds = [];
+
+const subscribeHook = async (z, bundle) => {
+	const { v4: uuidv4 } = require("uuid");
+	const zapier = require("zapier-platform-core");
+	zapier.tools.env.inject();
+	const eventIds = [];
+
+	const fetchEvents = {
```
Comment (sagarika-fynd) [inline triggers/createFile.js]: move this to utility file

File: triggers/createFile.js [hunk 19]
```diff
@@ -0,0 +1,252 @@
...
+		url: `${process.env.BASE_URL}/service/platform/notification/v1.0/events`,
+		method: "GET",
+	};
+
+	try {
+		let response = await z.request(fetchEvents);
+
+		if (response.status === 200) {
```
Comment (sagarika-fynd) [inline triggers/createFile.js]: name both temp and obj, correctly

File: triggers/createFile.js [hunk 20]
```diff
@@ -0,0 +1,252 @@
...
+	} catch (error) {
+		z.console.log("Error fetching events: " + error.message);
+		throw error;
+	}
+
+	const testWebHook = {
+		url: `${process.env.BASE_URL}/service/platform/notification/v1.0/webhook-configs/test`,
+		method: "POST",
+		body: {
+			url: "https://www.example.com",
```
Comment (sagarika-fynd) [inline triggers/createFile.js]: pass proper url

File: triggers/createFile.js [hunk 21]
```diff
@@ -0,0 +1,252 @@
...
+		let testHookResponse = await z.request(testWebHook);
+		if (testHookResponse.status === 200) {
+			try {
+				const webhookConfigResponse = await z.request({
+					url: `${process.env.BASE_URL}/service/platform/notification/v1.0/webhook-configs`,
+					method: "POST",
+					body: {
+						events: [...eventIds],
+						isActive: true,
+						name: uuidv4(),
```
Comment (sagarika-fynd) [inline triggers/createFile.js]: pre-pend with zap name

File: triggers/createFile.js [hunk 22]
```diff
@@ -0,0 +1,252 @@
...
+						url: bundle.targetUrl,
+					},
+				});
+
+				if (webhookConfigResponse.status === 200) {
+					hookID = webhookConfigResponse.data.webhookConfigId;
+					return webhookConfigResponse.data;
+				} else {
+					throw new Error(
+						`Failed to create webhook configuration. Status: ${webhookConfigResponse.status}`
```
Comment (sagarika-fynd) [inline triggers/createFile.js]: log the actuall error

File: triggers/createFile.js [hunk 23]
```diff
@@ -0,0 +1,252 @@
...
+		}
+	} catch (error) {
+		z.console.log("Error creating test WEBHOOK: " + error.message);
+		throw new Error(
+			`Failed to create a test webhook configuration. Status: ${error}`
+		);
+	}
+};
+
+const performList = async (z, bundle) => {
```
Comment (sagarika-fynd) [inline triggers/createFile.js:14]: write a comment to explain what is happening here

File: triggers/createFile.js [hunk 24]
```diff
@@ -0,0 +1,252 @@
...
+		public_id: `${process.env.CDN_URL}/v2/polished-hat-8f9bd4/original/dummy_image.png`,
+	};
+
+	let defaultPixelBinClient = new PixelbinClient(
+		new PixelbinConfig({
+			domain: `${process.env.BASE_URL}`,
+			apiSecret: bundle.authData.apiKey,
+		})
+	);
+	let temp = await defaultPixelBinClient.assets.listFilesPaginator({
```
Comment (sagarika-fynd) [inline triggers/createFile.js:81]: get only 1 file, this will return top 10 files

File: triggers/createFile.js [hunk 25]
```diff
@@ -0,0 +1,252 @@
...
+		method: "DELETE",
+	};
+
+	return z
+		.request(options)
+		.then((response) => {
+			if (response.status === 200) {
+				return response.data;
+			} else {
+				return unsubscribeHook(z, bundle, retries - 1);
```
Comment (sagarika-fynd) [inline triggers/createFile.js]: this will lead to infinite retry

File: triggers/createFile.js [hunk 26]
```diff
@@ -0,0 +1,252 @@
...
+		.catch((error) => {
+			if (retries > 0) {
+				return unsubscribeHook(z, bundle, retries - 1);
+			} else {
+				throw new Error(`Failed to delete after 5 attempts: ${error.message}`);
+			}
+		});
+};
+
+const deletePropertiesRecursive = (obj) => {
```
Comment (sagarika-fynd) [inline triggers/createFile.js:144]: just hard code this

File: triggers/createFile.js [hunk 27]
```diff
@@ -0,0 +1,252 @@
...
+		delete obj.querystring;
+		delete obj.s3Bucket;
+		delete obj.s3Key;
+
+		deletePropertiesRecursive(obj);
+	});
+
+	let obj = { ...bundle.cleanedRequest };
+
+	if (obj.event.name === "file") {
```
Comment (sagarika-fynd) [inline triggers/createFile.js:181]: remove it

File: triggers/createFile.js [hunk 28]
```diff
@@ -0,0 +1,252 @@
...
+		obj = {
+			...obj,
+			public_id: `${process.env.CDN_URL}/v2/${orgDetails?.org?.cloudName}/original/${obj.payload.fileId}`,
```
Comment (sagarika-fynd) [inline triggers/createFile.js:184]: get cdn url from appOrgDetails

File: triggers/createFile.js [hunk 29]
```diff
@@ -0,0 +1,252 @@
...
+		};
+	}
+
+	return [{ ...obj }];
+};
+
+module.exports = {
+	key: "createFile",
+
+	noun: "CreateFile",
```
Comment (sagarika-fynd) [inline triggers/createFile.js]: name it to file

File: triggers/createFile.js [hunk 30]
```diff
@@ -0,0 +1,252 @@
...
+	display: {
+		label: "Create File",
+		description: "Triggers when an image is uploaded to PixelBin.io.",
```
Comment (sagarika-fynd) [inline triggers/createFile.js:197]: remove full stop

File: triggers/createFolder.js [hunk 31]
```diff
@@ -0,0 +1,198 @@
+var hookID = "";
+const eventIds = [];
+
+const subscribeHook = async (z, bundle) => {
```
Comment (sagarika-fynd) [inline triggers/createFolder.js:1]: combine the code for createFolder, createFiler, and deleteFile

File: triggers/pixelbinEvents.js [hunk 32]
```diff
@@ -0,0 +1,383 @@
+var hookID = "";
+const eventIds = [];
+
+const subscribeHook = async (z, bundle) => {
```
Comment (sagarika-fynd) [inline triggers/pixelbinEvents.js]: rename this to PixelbinAllEvents.js, add comment for the same, with reason on why it is like this

File: triggers/pixelbinEvents.js [hunk 33]
```diff
@@ -0,0 +1,383 @@
...
+	const fetchEvents = {
+		url: `${process.env.BASE_URL}/service/platform/notification/v1.0/events`,
+		method: "GET",
+	};
+
+	try {
+		let response = await z.request(fetchEvents);
+
+		if (response.status === 200) {
+			const temp = [...response.data];
```
Comment (sagarika-fynd) [inline triggers/pixelbinEvents.js]: remove this filter for each of them , just give all

Comment (sagarika-fynd) [reply triggers/pixelbinEvents.js]: make it a dynamic field

File: build/build.zip
Comment (sagarika-fynd) [reply build/build.zip:1]: move to gitignore

**Review by sagarika-fynd — CHANGES_REQUESTED** [review]

File: triggers/createFolder.js (hunk 31 above)
Comment (sagarika-fynd) [reply triggers/createFolder.js:1]: Fetch hookID from bundle.subscribeData!.id
refer postHog :- 

https://github.com/PostHog/posthog-zapier/blob/main/src/utils.ts

File: README.md (hunk 1 above)
Comment (shantanuFyndGit) [reply README.md]: resolved

File: authentication.js (hunk 4 above)
Comment (shantanuFyndGit) [reply authentication.js]: resolved

File: triggers/createFile.js (hunk 29 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js]: resolved

File: triggers/createFile.js (hunk 30 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js:197]: resolved

File: build/build.zip
Comment (shantanuFyndGit) [reply build/build.zip:1]: resolved

File: build/source.zip
Comment (shantanuFyndGit) [reply build/source.zip:1]: resolved

File: creates/createReport.js
Comment (shantanuFyndGit) [reply creates/createReport.js:1]: resolved

File: creates/createReport.js (hunk 5 above)
Comment (shantanuFyndGit) [reply creates/createReport.js]: resolved

File: creates/createReport.js (hunk 6 above)
Comment (shantanuFyndGit) [reply creates/createReport.js]: resolved

File: creates/createReport.js (hunk 7 above)
Comment (shantanuFyndGit) [reply creates/createReport.js]: resolved

File: creates/transform.js (hunk 8 above)
Comment (shantanuFyndGit) [reply creates/transform.js]: resolved

File: creates/transform.js (hunk 9 above)
Comment (shantanuFyndGit) [reply creates/transform.js]: resolved

File: creates/transform.js (hunk 11 above)
Comment (shantanuFyndGit) [reply creates/transform.js]: resolved

File: creates/transform.js (hunk 13 above)
Comment (shantanuFyndGit) [reply creates/transform.js]: resolved

File: creates/upload.js (hunk 14 above)
Comment (shantanuFyndGit) [reply creates/upload.js]: resolved

File: .DS_Store
Comment (shantanuFyndGit) [reply .DS_Store:1]: resolved

File: creates/upload.js (hunk 15 above)
Comment (shantanuFyndGit) [reply creates/upload.js]: resolved

File: index.js (hunk 16 above)
Comment (shantanuFyndGit) [reply index.js:20]: resolved

Comment (shantanuFyndGit) [reply index.js:20]: not possible

File: package-lock.json
Comment (shantanuFyndGit) [reply package-lock.json:1]: resolved

File: samples/sample_repo_list.js (hunk 17 above)
Comment (shantanuFyndGit) [reply samples/sample_repo_list.js]: resolved

File: triggers/createFile.js (hunk 18 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js]: resolved

File: triggers/createFile.js (hunk 19 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js]: resolved

File: triggers/createFile.js (hunk 21 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js]: resolved

File: triggers/createFile.js (hunk 22 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js]: resolved

File: triggers/createFile.js (hunk 23 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js:14]: resolved

File: triggers/createFile.js (hunk 24 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js:81]: resolved

File: triggers/createFile.js (hunk 25 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js]: resolved

File: triggers/createFile.js (hunk 26 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js:144]: Need to be dynamic.

File: triggers/createFile.js (hunk 27 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js:181]: resolved

File: triggers/createFile.js (hunk 28 above)
Comment (shantanuFyndGit) [reply triggers/createFile.js:184]: Not getting into it

File: triggers/pixelbinEvents.js (hunk 32 above)
Comment (shantanuFyndGit) [reply triggers/pixelbinEvents.js]: resolved

File: triggers/pixelbinEvents.js (hunk 33 above)
Comment (shantanuFyndGit) [reply triggers/pixelbinEvents.js]: resolved

File: triggers/createFolder.js (hunk 31 above)
Comment (shantanuFyndGit) [reply triggers/createFolder.js:1]: resolved

File: authentication.js (hunk 2 above)
Comment (shantanuFyndGit) [reply authentication.js]: resolved

File: package.json [hunk 34]
```diff
@@ -0,0 +1,20 @@
+{
+	"name": "PixelBinMain",
```
Comment (sagarika-fynd) [inline package.json]: rename this as per standard pattern

File: samples/sample_issue.js [hunk 35]
```diff
@@ -0,0 +1,146 @@
+module.exports = {
+    "id": 1,
+    "url": "https://api.github.com/repos/octocat/Hello-World/issues/1347",
```
Comment (sagarika-fynd) [inline samples/sample_issue.js]: why have we not deleted this

File: test/authentication.test.js [hunk 36]
```diff
@@ -0,0 +1,37 @@
...
+
+const App = require("../index");
+const appTester = zapier.createAppTester(App);
+
+describe("custom auth", () => {
+	zapier.tools.env.inject();
+	it("passes authentication and returns json", async () => {
+		const bundle = {
+			authData: {
+				apiKey: "9410ee81-fcaa-4532-a18a-8d7bd4e3686c",
```
Comment (sagarika-fynd) [inline test/authentication.test.js]: delete this api-key

**Review by sagarika-fynd — CHANGES_REQUESTED** [review]

File: package.json (hunk 34 above)
Comment (shantanuFyndGit) [reply package.json]: resolved

File: test/authentication.test.js (hunk 36 above)
Comment (shantanuFyndGit) [reply test/authentication.test.js]: resolved

**Review by sagarika-fynd — APPROVED** [review]
